core/app/models/spree/store.rb

```diff
@@ -1,5 +1,8 @@
 module Spree
   class Store < Spree::Base
+    MAILER_LOGO_CONTENT_TYPES = ['image/png', 'image/jpg', 'image/jpeg'].freeze
+    FAVICON_CONTENT_TYPES = ['image/png', 'image/x-icon', 'image/vnd.microsoft.icon'].freeze
```

Comment:
You could probably get away with adding support for SVG to `FAVICON_CONTENT_TYPES`, although its not 100% supported by all browsers. https://caniuse.com/link-icon-svg